### PR TITLE
Revise dropdowns

### PR DESCRIPTION
**Pull request by 70ray**
Use addEventListener instead of onclick to toggle dropdown menus. For issue #1298

Sandbox at: https://www.pgdp.org/~rp31/c.branch/revise_dropdowns

### CODE REVIEW

File: tools/dropdown.js [hunk 1]
```diff
@@ -1,40 +1,44 @@
...
-    var thisDropdown = document.getElementById(id);
-    var turnOn = !thisDropdown.classList.contains("dropdown-show");
-    closeAll();
-    if (turnOn) {
-        thisDropdown.classList.add("dropdown-show");
+    function closeAll() {
+        var dropdowns = document.getElementsByClassName("dropdown-content");
+        for (let openDropdown of dropdowns) {
+            if (openDropdown.classList.contains("dropdown-show")) {
+                openDropdown.classList.remove("dropdown-show");
```
Comment (chrismiceli) [inline tools/dropdown.js:13]: It doesn't seem you need the if at all: https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/remove
```
 If the string is not in the list, no error is thrown, and nothing happens. 
```

File: tools/dropdown.js [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
+        }
     }
-}
 
-// Close the dropdown if the user clicks outside of it
-window.onclick = function (event) {
-    "use strict";
-    if (!event.target.matches(".dropdown-button")) {
+    function toggleList() {
+        var thisDropdown = this.nextElementSibling;
```
Comment (chrismiceli) [inline tools/dropdown.js:19]: can we use const instead of var?

File: tools/dropdown.js (hunk 1 above)
Comment (70ray) [reply tools/dropdown.js:13]: I've removed it.

File: tools/dropdown.js (hunk 2 above)
Comment (70ray) [reply tools/dropdown.js:19]: yes, I've changed all the vars to consts

**Comment by 70ray** [comment]
Sandbox updated.

**Review by jchaffraix — APPROVED** [review]

**Review by chrismiceli — APPROVED** [review]

File: tools/dropdown.js [hunk 3]
```diff
@@ -1,40 +1,44 @@
...
-    var dropdowns = document.getElementsByClassName("dropdown-content");
-    var i, openDropdown;
-    for (i = 0; i < dropdowns.length; i++) {
-        openDropdown = dropdowns[i];
-        if (openDropdown.classList.contains("dropdown-show")) {
-            openDropdown.classList.remove("dropdown-show");
-        }
-    }
-}
+"use strict";
```
[the remark this answers is not in the document]
Comment (srjfoo) [reply tools/dropdown.js:1]: Did you decide not to move it to the scripts folder? (Or maybe I'm missing something, but I still see it in `tools`.

**Review by srjfoo — APPROVED** [review]

File: tools/dropdown.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (70ray) [reply tools/dropdown.js:1]: I was waitig to see what @cpeel thought about it.

[the remark this answers is not in the document]
Comment (cpeel) [reply tools/dropdown.js:1]: Whoops, sorry - I agree moving this to `scripts/` makes sense to me.

**Comment by 70ray** [comment]
Sandbox updated.

File: tools/dropdown.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (70ray) [reply tools/dropdown.js]: Moved it now.

**Review by chrismiceli — APPROVED** [review]

**Review by cpeel — APPROVED** [review]

**Pull request merged**